share/commands/server-upgrade-techniques

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+# @description upgrade techniques in the configuration repository from the packaged ones
+# @man This command will replace the techniques in /var/rudder/configuration-repository/techniques
+# @man by the techniques found in /opt/rudder/share/techniques which is installed by rudder-technique package.
+# @man The upgrade can take care of used defined changes if a proper tag has been set during previous upgrade.
+# @man +
+# @man *Options*:
+# @man +
+# @man *-o*: override existing technique without looking for local changes
+# @man +
+# @man *-t*: do not upgrade, but tag the repository to make next upgrade easier
+# @man +
+# @man *-c*: take the given commit id as a starting point for merging
+# @man +
+
+. "${BASEDIR}/../lib/common.sh"
+. "${BASEDIR}/../lib/api_call.sh"
+
+DISPLAY_COMMAND=false
+
+while getopts "otc:" opt; do
+  case $opt in
+    o)
+      OVERRIDE=true
+      ;;
+    t)
+      TAG=true
+      ;;
+    c)
+      COMMIT_ID=${OPTARG}
+      ;;
+  esac
+done
+
+REPO="/var/rudder/configuration-repository"
+BASE="/opt/rudder/share/techniques"
+TAG_MESSAGE="-upgrade-"
+if [ "${OVERRIDE}" = "true" ]
+then
+  cd "${REPO}"
+  cp -a "${BASE}"/* techniques/
+  git add techniques/
+  git commit -q -m "${TAG_MESSAGE} Forced technique upgrade from version ${package_version} on $(date)"
+  rudder server reload-techniques
+elif [ "${TAG}" = "true" ]
+then
+  cd "${REPO}"
+  git commit --allow-empty -q -m "${TAG_MESSAGE} No technique upgrade on $(date)"
+else
+  if [ "${COMMIT_ID}" = "" ]
+  then
+    commit=$(git log --oneline | grep "^[0-9a-f]* ${TAG_MESSAGE}" | head -n1)
+    if [ "${commit}" = "" ]
+    then
+      echo "Upgrade without override not supported without tag message!"
```

Comment:
WHy is that impossible ?

Reply:
WHy not starting from HEAD ?  

Why not creating the branch from the commit and not HEAD ?

Reply:
ping @peckpeck ?

Reply:
Without a tag message we don't know where to start from. There is not commit id to start from.
If someone has touched the techniques the content will be overridden, that's why the user must be aware of this and use the override option.